tests/testthat/test-plot_step_ahead_model_output.R

```diff
@@ -204,6 +208,45 @@ test_that("Output", {
   expect_equal(strsplit(strsplit(tail(purrr::map(plot_test$x$attrs,
                                                  "hovertext"), 1)[[1]],
                                  "<br>")[[1]][2], "Intervals")[[1]][1], "90% ")
+
+  ## Interactive plot with 3 model, 1 model did not submit one scenario
+  test_data <- dplyr::filter(projection_data,
+                             !(model_id == "HUBuni-simexamp" &
+                                 scenario_id == "A-2021-03-05"))
+  plot_test <-
+    plot_step_ahead_model_output(test_data,
+                                 target_data_us,
+                                 use_median_as_point = TRUE,
+                                 facet = "scenario_id")
+  leg_sel <- !unlist(purrr::map(purrr::map(plot_test$x$data, "showlegend"),
+                                isFALSE))
+  legend <- unique(unlist(purrr::map(purrr::map(plot_test$x$data,
+                                                "name")[leg_sel],
+                                     as.character)))
```

Comment:
You can use the underscore functions to extract these values instead of a double-map. This makes the code a bit easier to reason about when coming back to it. 

```suggestion
  leg_sel <- purrr::map_lgl(plot_test$x$data, "showlegend")
  legend <- purrr::map_chr(plot_test$x$data, \(x) as.character(x$name))[leg_sel]
```